crates/metaslang/cst/src/query/parser.rs

```diff
@@ -26,10 +26,28 @@ use crate::text_index::{TextIndex, TextRange};
 /// Error from parsing a query.
 #[derive(Clone, Debug, Error)]
 pub struct QueryError {
+    message: String,
+    text_range: TextRange,
+}
+
+impl QueryError {
     /// The error message describing the parsing issue.
-    pub message: String,
+    pub fn message(&self) -> &str {
+        &self.message
+    }
+
     /// The range of text where the error occurred.
-    pub text_range: TextRange,
+    pub fn text_range(&self) -> &TextRange {
+        &self.text_range
+    }
+
+    /// Creates a new `QueryError` with the given message and text range.
+    pub fn create(message: String, text_range: TextRange) -> Self {
```

Comment:
Actually, shouldn't this be hidden to the user? Why would a user need to create a QueryError?

Reply:
While not expected to be used by external users, it is used by other crates (like `metaslang_graph_builder`) to construct query errors.

Reply:
It is used by other crates (like metaslang_graph_builder) to construct query errors.
While not expected to be used by external users for most use cases, it can still be useful to create errors for testing/comparisons..